src/infrastructure/Models/ValidationRulesTranslator.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Infrastructure\Models;
+
+use Doctrine\Common\Annotations\AnnotationException;
+use Doctrine\Common\Annotations\AnnotationReader;
+use Infrastructure\Annotations\Validation;
+use Infrastructure\Exceptions\InfrastructureException;
+use Infrastructure\Services\BaseService;
+use ReflectionClass;
+use ReflectionException;
+use Symfony\Component\Validator\Constraints\Date;
+use Symfony\Component\Validator\Constraints\DateTime;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\Time;
+use Symfony\Component\Validator\Constraints\Type;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\InvalidOptionsException;
+use Symfony\Component\Validator\Exception\MissingOptionsException;
+
+class ValidationRulesTranslator
+{
+    /**
+     * @param array $ruleDescriptions
+     * @return array
+     * @throws ConstraintDefinitionException
+     * @throws InvalidOptionsException
+     * @throws MissingOptionsException
+     */
+    public function translate(array $ruleDescriptions) : array
+    {
+        $rules = [];
+
+        /** @var Validation $ruleDescription */
+        foreach ($ruleDescriptions as $ruleDescription) {
+
+            if (!($ruleDescription instanceof Validation)){
+                continue;
+            }
+
+            $rule = new ValidationRule($ruleDescription->name);
+
+            if (!$ruleDescription->type) {
+                $ruleDescription->type = 'string';
+            }
+
+            foreach ($this->constrainsMap() as $field => $constrain) {
+
+                if (!$ruleDescription->$field){
+                    continue;
+                }
+
+                $rule->addConstraint($constrain($ruleDescription->$field));
+            }
+
+            $rules[] = $rule;
+        }
+
+        return $rules;
+    }
+
+    /**
+     * @param $type
+     * @return Type
+     * @throws InfrastructureException
+     * @throws ConstraintDefinitionException
+     * @throws InvalidOptionsException
+     * @throws MissingOptionsException
+     */
+    public function addType($type) : Type
```

Comment:
why this method is  public? Do you use somewhere else?

Reply:
fixed